File: LMIS/LMIS.Web/Modules/ReaderManage/CardLevel/CardLevelForm.cs
using Serenity.ComponentModel;
using System;

namespace LMIS.ReaderManage.Forms;

[FormScript("ReaderManage.CardLevel")]
[BasedOnRow(typeof(CardLevelRow), CheckNames = true)]
public class CardLevelForm
{
    public string CardLevelName { get; set; }
    [DecimalEditor]
    public decimal Fee { get; set; }
    [DecimalEditor]
    public decimal Rent { get; set; }
    [DecimalEditor]
    public decimal Discount { get; set; }
    [DecimalEditor]
    public decimal Penalty { get; set; }
    [IntegerEditor]
    public int BorrowDays { get; set; }
    [IntegerEditor]
    public int BorrowBooks { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/CardLevelForm.ts
import { StringEditor, DecimalEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CardLevelForm {
    CardLevelName: StringEditor;
    Fee: DecimalEditor;
    Discount: DecimalEditor;
    Penalty: DecimalEditor;
    BorrowDays: IntegerEditor;
    BorrowBooks: IntegerEditor;
}

export class CardLevelForm extends PrefixedContext {
    static readonly formKey = 'ReaderManage.CardLevel';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CardLevelForm.init)  {
            CardLevelForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = IntegerEditor;

            initFormType(CardLevelForm, [
                'CardLevelName', w0,
                'Fee', w1,
                'Discount', w1,
                'Penalty', w1,
                'BorrowDays', w2,
                'BorrowBooks', w2
            ]);
        }
    }
}